cpp/arcticdb/column_store/column_data.hpp

```diff
@@ -217,8 +219,7 @@ struct ColumnData {
         using base_type = base_iterator_type<TDT, iterator_type, iterator_density, constant>;
         using RawType = typename TDT::DataTypeTag::raw_type;
     public:
-        ColumnDataIterator() = delete;
-
+        ColumnDataIterator() = default;
```

Comment:
This was intentional change. If `ColumnDataIterator` has a default constructor it implements the concept needed by most of the algorithms and classes in `std::ranges`. I used this in particular to create a `std::ranges::subrange` in `sorted_aggregation.cpp` and avoid passing iterators around.